kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebsocketKieServerClient.java

```diff
@@ -0,0 +1,524 @@
+            ServiceResponsesList list = WebsocketUtils.unmarshal(message, MarshallingFormat.JSON.getType(), ServiceResponsesList.class);            
+            return list.getResponses().get(0);
+        })).getResponses().get(0);
+        
+        return response; 
+    }
+
+    @Override
+    public ServiceResponse<ReleaseId> getReleaseId(String containerId) {
+        CommandScript script = new CommandScript(Collections.singletonList(new GetReleaseIdCommand(containerId)));
+        ServiceResponse<ReleaseId> response = (ServiceResponse<ReleaseId>) sendCommand(script, new WebsocketServiceResponse(true, (message) -> {
+            ServiceResponsesList list = WebsocketUtils.unmarshal(message, MarshallingFormat.JSON.getType(), ServiceResponsesList.class);            
+            return list.getResponses().get(0);
+        })).getResponses().get(0);
+        
+        return response; 
+    }
+
+    @Override
+    public ServiceResponse<ReleaseId> updateReleaseId(String id, ReleaseId releaseId) {
+        CommandScript script = new CommandScript(Collections.singletonList((KieServerCommand) new UpdateReleaseIdCommand(id, releaseId)));
+        ServiceResponse<ReleaseId> response = (ServiceResponse<ReleaseId>) sendCommandToAllSessions(script, new WebsocketServiceResponse(true, (message) -> {
+            ServiceResponsesList list = WebsocketUtils.unmarshal(message, MarshallingFormat.JSON.getType(), ServiceResponsesList.class);            
+            return list.getResponses().get(0);
+        })).getResponses().get(0);
+        
+        return response; 
+    }
+
+    @Override
+    public ServiceResponse<KieServerStateInfo> getServerState() {
+        CommandScript script = new CommandScript(Collections.singletonList((KieServerCommand) new GetServerStateCommand()));
+        ServiceResponse<KieServerStateInfo> response = (ServiceResponse<KieServerStateInfo>) sendCommand(script, new WebsocketServiceResponse(true, (message) -> {
+            ServiceResponsesList list = WebsocketUtils.unmarshal(message, MarshallingFormat.JSON.getType(), ServiceResponsesList.class);            
+            return list.getResponses().get(0);
+        })).getResponses().get(0);
+        
+        return response; 
+    }
+
+    @Override
+    public ServiceResponse<String> executeCommands(String id, String payload) {
+        throw new UnsupportedOperationException("Not supported for websocket implementation");
+    }
+
+    @Override
+    public ServiceResponse<String> executeCommands(String id, Command<?> cmd) {
+        throw new UnsupportedOperationException("Not supported for websocket implementation");
+    }
+
+    @Override
+    public void setClassLoader(ClassLoader classLoader) {
+        // no-op
+    }
+
+    @Override
+    public ClassLoader getClassLoader() {
+        return this.getClassLoader();
```

Comment:
This method recursively calls itself, which will result in stackoverflow. Is that ok?